sdk/src/manifest.rs

```diff
@@ -2209,6 +2223,33 @@ pub(crate) mod tests {
         assert!(active_manifest.thumbnail().is_none());
     }
 
+    #[test]
+    fn test_missing_thumbnail() {
+        const MANIFEST_JSON: &str = r#"
+            {
+                "claim_generator": "test",
+                "format" : "image/jpeg",
+                "thumbnail": {
+                    "format": "image/jpeg",
+                    "identifier": "does_not_exist.jpg"
+                }
+            }
+        "#;
+
+        let mut manifest = Manifest::from_json(MANIFEST_JSON).expect("from_json");
+
+        let mut source = std::io::Cursor::new(vec![1, 2, 3]);
+        let mut dest = std::io::Cursor::new(Vec::new());
+        let signer = temp_signer();
+        let result =
+            manifest.embed_to_stream("image/jpeg", &mut source, &mut dest, signer.as_ref());
+        assert!(result.is_err());
+        assert!(result
+            .unwrap_err()
+            .to_string()
+            .contains("resource not found: does_not_exist.jpg"));
```

Comment:
Not required for merging, but I've been using this pattern to test for errors so we don't have to test against the error strings of the c2pa-rs error. 

```suggestion
        assert!(matches!(
            result,
            Err(crate::Error::ResourceNotFound(_)
        ))
```

Reply:
I'm going to stick with this test here, because I'm trying to validate the filename inside the error as well as the error type. But if I was just validating for the type of error, I agree the other is better.